akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -662,23 +661,6 @@ private[akka] class Shard(
       }
     case RestartRememberedEntities(entities) => restartEntities(entities)
     case l: LeaseLost                        => receiveLeaseLost(l)
-    case ShardRegion.StartEntityAck(entityId, _) =>
-      if (update.started.contains(entityId)) {
-        // currently in progress of starting, so we'll need to stop it when that is done
-        entities.rememberingStop(entityId, StartedElsewhere)
-      } else {
-        entities.entityState(entityId) match {
-          case _: RememberingStart =>
-            // queued up for batched start, let's just not start it
-            entities.removeEntity(entityId)
-          case _: Active =>
-            // add to stop batch
-            entities.rememberingStop(entityId, StartedElsewhere)
```

Comment:
As far as I can see, this can never happen, if the shard id extractor was changed, the entity will never end up started here.